RealmSwift/Impl/ComplexTypes.swift

```diff
@@ -83,6 +83,33 @@ extension EmbeddedObject: SchemaDiscoverable, _PersistableInsideOptional, _Defau
     }
 }
 
+extension AsymmetricObject: SchemaDiscoverable, _PersistableInsideOptional, _DefaultConstructible {
+    public typealias PersistedType = AsymmetricObject
+    public static var _rlmType: PropertyType { .object }
+    public static func _rlmPopulateProperty(_ prop: RLMProperty) {
+        Object._rlmPopulateProperty(prop)
+        prop.objectClassName = className()
+    }
+
+    public static func _rlmGetProperty(_ obj: ObjectBase, _ key: UInt16) -> Self {
```

Comment:
Similarly these don't apply to asymmetric objects, as you can't read one from the Realm.

The schema discovery parts do apply, though.

Reply:
I guess we don't even need to conform to SchemaDiscovery as we only need to know if the table is asymmetric and that comes from the override `isAsymmetric`, or am I missing something?. I removed the conformance and everything works fine.

Reply:
No, you're right and we don't need the schema discovery part either.